src/shared/auth.ts

```diff
@@ -16,6 +18,13 @@ export const authOptions: NextAuthOptions = {
   pages: {
     signIn: "/login",
   },
+  callbacks: {
+    async signIn(user) {
+      const loginUser = user.user as AdapterUser;
+      createUser(loginUser);
+      return true;
```

Comment:
You can use a try catch here.

if `try` block works, return true. If the code catch an error, you can return false